interfaces/builtin/jack.go

```diff
@@ -0,0 +1,52 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const jackSummary = `allows access to the jack socket`
+
+const jackBaseDeclarationSlots = `
+  jack:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+/*
+ * libjack builds the shared memory endpoint like this :
+ * jack-<userid>/<server name>/jack-<server id>
+ *
+ * see libjack/shm.c in jack1's source tree,
+ * and common/shm.c in jack2's source tree.
+ */
+const jackConnectedPlugAppArmor = `
+owner /dev/shm/jack-[0-9]*/*/* rw,
+`
```

Comment:
Should the jack interface carry system calls for connecting to the socket? I understand those are easily granted by network but perhaps they should be explicitly allowed here as well?

Reply:
I will need help on this, as I'm really not an expert. Do you mean I should add "shmctl", as in pulseaudio.go ?

So far in my test snap I only have the `jack` plug, and it works fine :

```
apps:
  jack-client-test:
    command: jack-client-test
    plugs:
      - jack
```

Reply:
I think we should have enough permissions to open the socket and talk to it. This includes things like calling `socket` and `connect` with appropriate arguments. Perhaps one thing that would help to flesh this out would be a simple test that shows one can indeed talk to the Jack socket?

A trivial C program that opens the socket and asks for, say, protocol version, or something equivalent, would be beneficial to the confidence in completeness of this pull request.

Reply:
I believe you have everything you need in the pull request description. [This is the snap I use](https://github.com/frenchguych/jack-test/) to test my changes on my machine. It connects to the jack server and emits a beep every second.
If that is not clear or not enougth, I'd be happy to provide another test case.

Reply:
Ah, I missed that, sorry. I'm somewhat used to having terse description where all the interesting things are in the pull request diff. I will re-process this with the test snap in place. Thanks!

Reply:
> Should the jack interface carry system calls for connecting to the socket? I understand those are easily granted by network but perhaps they should be explicitly allowed here as well?

See my comment above. Nothing else is needed beyond what is already granted in the default template.